.circleci/workflows.yml

```diff
@@ -575,13 +570,17 @@ workflows:
           requires:
             - compile-contracts
             - create-relayers
-      - build-and-publish-container-image:
+      - test-container-image:
+          requires:
+            - compile-contracts
+            - create-relayers
+      - publish-container-image:
```

Comment:
I would actually move this (`publish-container-image`) out into its own workflow, and make it depend on all the other workflows.

The way this is now, we could theoretically publish a container image even if there is a failure in one of the agent and contract tests, which is not nice. Moving the publish part into its own workflow and making it depend on all tests should solve that.